menas/src/test/scala/za/co/absa/enceladus/menas/schema/SchemaConvertersSuite.scala

```diff
@@ -0,0 +1,412 @@
+        |								"fields": [
+        |									{
+        |										"name": "field11",
+        |										"type": "date",
+        |										"nullable": true,
+        |										"metadata": {
+        |											"pattern": "yyyy-MM-dd"
+        |										}
+        |									},
+        |									{
+        |										"name": "field12",
+        |										"type": "timestamp",
+        |										"nullable": true,
+        |										"metadata": {
+        |											"pattern": "yyyy-MM-dd'T'HH:mm:ss"
+        |										}
+        |									},
+        |									{
+        |										"name": "field13",
+        |										"type": "integer",
+        |										"nullable": false,
+        |										"metadata": {}
+        |									},
+        |									{
+        |										"name": "field14",
+        |										"type": "string",
+        |										"nullable": true,
+        |										"metadata": {}
+        |									}
+        |								]
+        |							},
+        |							"containsNull": true
+        |						},
+        |						"nullable": true,
+        |						"metadata": {}
+        |					}
+        |				]
+        |			},
+        |			"nullable": true,
+        |			"metadata": {}
+        |		},
+        |		{
+        |			"name": "field15",
+        |			"type": "string",
+        |			"nullable": true,
+        |			"metadata": {}
+        |		}
+        |	]
+        |}
+      """.stripMargin
+
+    val expectedSchema = StructType(Seq(
+      StructField("field1",StringType,true),
+      StructField("field2",IntegerType,true),
+      StructField("field3",DateType,true,Metadata.fromJson("""{"pattern": "yyyy-MM-dd"}""")),
+      StructField("field4",DecimalType(38,18),true),
+      StructField("field5",BooleanType,true),
+      StructField("field6",TimestampType,true,Metadata.fromJson("""{"pattern": "yyyy-MM-dd'T'HH:mm:ss"}""")),
+      StructField("nested1",StructType(Seq(
+        StructField("nested2",ArrayType(StructType(Seq(
+          StructField("field7",DecimalType(38,18),true),
+          StructField("field8",StringType,false),
+          StructField("field9",DateType,true,Metadata.fromJson("""{"pattern": "yyyy-MM-dd"}""")),
+          StructField("field10",IntegerType,true))),true),true)
+      )),true),
+      StructField("nested3",StructType(Seq(
+        StructField("nested4",ArrayType(StructType(Seq(
+          StructField("field11",DateType,true,Metadata.fromJson("""{"pattern": "yyyy-MM-dd"}""")),
+          StructField("field12",TimestampType,true,Metadata.fromJson("""{"pattern": "yyyy-MM-dd'T'HH:mm:ss"}""")),
+          StructField("field13",IntegerType,false),
+          StructField("field14",StringType,true))),true),true))),true),
+      StructField("field15",StringType,true)
+    ))
+
+    val actualSchema = sparkConverter.convertAnyToStructType(inputJson)
+
+    assert(actualSchema.prettyJson == expectedSchema.prettyJson)
+  }
+
+  test("Test 0.9.* Menas to StructType schema conversion") {
```

Comment:
I would keep this version number out of the project, just call it `alpha Menas` or `pre-release Menas`.

Reply:
yeah... it is actually 0.4.180. Not sure if 'alpha' is a proper name though, it is 1 year in prod.